x/interchainstaking/keeper/ibc_packet_handlers.go

```diff
@@ -830,7 +830,12 @@ func (k *Keeper) prepareRewardsDistributionMsgs(zone types.Zone, rewards sdk.Coi
 	var msgs []sdk.Msg
 
 	dust := rewards.Amount
-	portion := rewards.Amount.ToDec().Quo(sdk.NewDec(int64(len(zone.DelegationAddresses)))).TruncateInt()
+	numDelegators := len(zone.DelegationAddresses)
+	if numDelegators == 0 {
+		// zone has no delegators, we must panic here, something is very wrong...
+		panic("internal zone error, zone has no delegators")
```

Comment:
@zanicar - I think this can plausibly happen if the zone has only just been registered and an epoch hits. We should throw and error here, and catch it when iterating the zones, and just continue to the next.